src/app/core/oauth/oauth.service.ts

```diff
@@ -49,150 +54,78 @@ export class OauthService {
     if (this.oauthSectionDeclared) {
       return this.requestInfoSubject
     }
 
-    // TODO @leomendoza123 @DanielPalafox remove mock data and use the following function
-
-    // return this._http
-    //   .post<RequestInfoForm>(
-    //     environment.BASE_URL + '/oauth/custom/declare.json',
-    //     value,
-    //     { headers: this.headers }
-    //   )
-    //   .pipe(
-    //     retry(3),
-    //     catchError((error) => this._errorHandler.handleError(error))
-    //   )
-    //   .pipe(
-    //     tap((data) => {
-    //       this.requestInfoSubject.next(data)
-    //       this.oauthSectionDeclared = true
-    //     })
-    //   )
+    return this._http
+      .post<RequestInfoForm>(
+        // tslint:disable-next-line:max-line-length
+        environment.BASE_URL +
+          `oauth/custom/init.json?client_id=${value.client_id}&response_type=${value.response_type}&scope=${value.scope}&redirect_uri=${value.redirect_uri}`,
+        value,
+        { headers: this.headers }
+      )
+      .pipe(
+        retry(3),
+        catchError((error) => this._errorHandler.handleError(error))
+      )
+      .pipe(
+        tap((data) => {
+          this.requestInfoSubject.next(data)
+          this.oauthSectionDeclared = true
+        })
+      )
+  }
 
-    return of(<RequestInfoForm>{
-      errors: [],
-      scopes: [
-        {
-          name: 'AUTHENTICATE',
-          value: '/authenticate',
-          description: 'Get your ORCID iD',
-          longDescription:
-          // tslint:disable-next-line: max-line-length
-            'Allow this organization or application to get your 16-character ORCID iD and read information on your ORCID record you have marked as public.',
-        },
-        {
-          name: 'OPENID',
-          value: 'openid',
-          description: 'Get your ORCID iD',
-          longDescription:
-          // tslint:disable-next-line: max-line-length
-            'Allow this organization or application to get your 16-character ORCID iD and read information on your ORCID record you have marked as public.',
-        },
-      ],
-      clientDescription: 'https://developers.google.com/oauthplayground\t',
-      clientId: 'APP-MLXS7JVFJS9FEIFJ',
-      clientName: 'test',
-      clientEmailRequestReason: '',
-      memberName: 'asda',
-      redirectUrl: 'https://developers.google.com/oauthplayground?code=1R3eAY',
-      responseType: 'code',
-      stateParam: null,
-      userId: null,
-      userName: 'logedINTEST',
-      userOrcid: 'logedInTest',
-      userEmail: null,
-      userGivenNames: null,
-      userFamilyNames: null,
-      nonce: null,
-      clientHavePersistentTokens: true,
-      scopesAsString: '/authenticate openid',
-    }).pipe(
-      tap((data) => {
-        this.requestInfoSubject.next(data)
-        this.oauthSectionDeclared = true
-      })
-    )
+  oauthAuthorize(value: DeclareOauthSession): Observable<RequestInfoForm> {
```

Comment:
let's make this method name more explicit so is easier to understand what it does. Maybe something like `updateOauthSession`

Also, this function itself can be the one updating the `requestInfoSubject`, instead of depending on a separate method that required to be externally called.

Reply:
Done!

Reply:
Ok